crates/ra_proc_macro_srv/src/cli.rs

```diff
@@ -4,15 +4,13 @@ use crate::{expand_task, list_macros};
 use ra_proc_macro::msg::{self, Message};
 use std::io;
 
-pub fn run() {
+pub fn run() -> io::Result<()> {
     loop {
-        let req = match read_request() {
-            Err(err) => {
-                eprintln!("Read message error on ra_proc_macro_srv: {}", err);
-                continue;
-            }
-            Ok(None) => continue,
-            Ok(Some(req)) => req,
+        // bubble up the error for read request,
+        // as the stdin pipe may be closed.
+        let req = match read_request()? {
```

Comment:
These can now be re-written as `while let Some(req) = read_request()?`. Also, I'd personally would remove the comment -- I think the `?` and `io` error are pretty self-explanatory. 

bors d+